src/date/date.c

```diff
@@ -1,6 +1,338 @@
+#define _POSIX_C_SOURCE 200112L // for setenv
```

Comment:
You will need to create your own `setenv`. Fear not, it is simpler than it seems.
Take a look at [glibc implementaion](https://sourceware.org/git/?p=glibc.git;a=blob;f=stdlib/setenv.c;h=e2164371ade896e9b3342754724e5ca381beb64e;hb=HEAD) or [muscl one](https://git.musl-libc.org/cgit/musl/tree/src/env/setenv.c)